File: backend/models/subscription.py
from pydantic import BaseModel, Field
from typing import Optional
from datetime import datetime

class SubscriptionStatus(BaseModel):
    subscription_status: str  # 'trial', 'active', 'expired', 'lifetime'
    trial_ends_at: Optional[datetime] = None
    plan_expiry_date: Optional[datetime] = None
    plan_type: Optional[str] = None  # 'monthly', 'yearly', 'lifetime'
    days_remaining: int = 0
    is_trial_expired: bool = False
    is_active: bool = False
    can_perform_action: bool = True

class CreateOrderRequest(BaseModel):
    amount_in_paise: int
    plan_id: str  # 'monthly' or 'yearly'

class CreateOrderResponse(BaseModel):
    order_id: str
    razorpay_key: str
    amount: int
    currency: str = "INR"

class CompletePaymentRequest(BaseModel):
    razorpay_payment_id: str
    razorpay_order_id: str
    razorpay_signature: str
    plan_id: str  # 'monthly' or 'yearly'

class RestoreSubscriptionRequest(BaseModel):
    pass  # Empty, will use current user from auth

class SubscriptionUpdateRequest(BaseModel):
    subscription_status: str
    plan_type: str
    plan_expiry_date: datetime

File: src/api/subscription.ts
import apiClient from './client';

export interface SubscriptionStatus {
  subscription_status: string;
  trial_ends_at: string | null;
  plan_expiry_date: string | null;
  plan_type: string | null;
  days_remaining: number;
  is_trial_expired: boolean;
  is_active: boolean;
  can_perform_action: boolean;
}

export interface CreateOrderRequest {
  amount_in_paise: number;
  plan_id: string;
}

export interface CreateOrderResponse {
  order_id: string;
  razorpay_key: string;
  amount: number;
  currency: string;
}

export interface CompletePaymentRequest {
  razorpay_payment_id: string;
  razorpay_order_id: string;
  razorpay_signature: string;
  plan_id: string;
}

export const subscriptionApi = {
  getStatus: async (): Promise<SubscriptionStatus> => {
    const response = await apiClient.get<SubscriptionStatus>('/subscription/status');
    return response.data;
  },

  createOrder: async (data: CreateOrderRequest): Promise<CreateOrderResponse> => {
    const response = await apiClient.post<CreateOrderResponse>('/subscription/create-order', data);
    return response.data;
  },

  completePayment: async (data: CompletePaymentRequest): Promise<SubscriptionStatus> => {
    const response = await apiClient.post<SubscriptionStatus>('/subscription/complete', data);
    return response.data;
  },

  restore: async (): Promise<SubscriptionStatus> => {
    const response = await apiClient.post<SubscriptionStatus>('/subscription/restore', {});
    return response.data;
  },
};
